### PR TITLE
manage_special_days.php: Use DPDatabase APIs, sprintf, set_col_str

### PR DESCRIPTION
**Pull request by bpfoley**
Move set_col_num, set_col_str to misc because it's generally useful.

### CODE REVIEW

**Comment by bpfoley** [comment]
Available in https://www.pgdp.org/~bfoley/c.branch/cleanup15/

No obvious problems adding, removing, editing a special day.

More needs to be done here to clean up the input handling rather than using raw `$_POST`, but I'm trying to limit the scope of these changes.

File: pinc/misc.inc [hunk 1]
```diff
@@ -596,21 +620,20 @@ function surround_and_join( $strings, $L, $R, $joiner )
...
 {
-    $update_fields = array();
+    $update_fields = [];
     foreach($source_array as $field => $value)
     {
-        if(in_array($field, $string_fields))
+        if (in_array($field, $string_fields))
         {
-            $value = sprintf("'%s'", mysqli_real_escape_string(DPDatabase::get_connection(), trim($value)));
+	    $update_fields[] = set_col_str($field, $value);
```
Comment (cpeel) [inline pinc/misc.inc:628]: We should retain the `trim()` or we should ensure that this is done (if it needs to be done) in all of the callers of this function.

**Comment by srjfoo** [comment]
> No obvious problems adding, removing, editing a special day.

No problem adding or editing ... but the script doesn't remove special days, does it? (enable and disable, yes, but not actually delete)

**Comment by bpfoley** [comment]
> > No obvious problems adding, removing, editing a special day.
> 
> No problem adding or editing ... but the script doesn't remove special days, does it? (enable and disable, yes, but not actually delete)

Yup, you're right. There's no actual delete (which is a little odd, when you think about it.)

**Comment by srjfoo** [comment]
> > > No obvious problems adding, removing, editing a special day.
> > 
> > 
> > No problem adding or editing ... but the script doesn't remove special days, does it? (enable and disable, yes, but not actually delete)
> 
> Yup, you're right. There's no actual delete (which is a little odd, when you think about it.)

As far as I can tell, we don't delete special days. We may disable them, but if a project was ever created and made it all the way through with the special day still enabled, it's associated with that project forever. If we deleted special days, we'd have to go through and make sure that any project that used it had it removed.

**Review by srjfoo — APPROVED** [review]
Works as expected as far as I can tell.

**Pull request merged**